models/DataObject/Concrete.php

```diff
@@ -624,13 +634,13 @@ public static function __callStatic($method, $arguments)
                     throw new \Exception('Call to undefined static method ' . $method . ' in class DataObject\\Concrete');
                 }
 
-                if (!in_array($localizedField->getFieldType(), $allowedDataTypes)) {
-                    throw new \Exception("Static getter '::getBy".ucfirst($propertyName)."' is not allowed for fieldtype '" . $localizedField->getFieldType() . "', it's only allowed for the following fieldtypes: " . implode(',', $allowedDataTypes));
+                if ($localizedField->isFilterable()) {
```

Comment:
I think, it should be `if (!$localizedField->isFilterable()) {` right?

Reply:
You are right, thanks. Fixed this.